qubo-lifecycle.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2020 D-Wave Systems Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+from collections import defaultdict
+from dimod import BinaryQuadraticModel as BQM
+import networkx as nx
+import dwave.embedding
+from dwave.system import DWaveSampler, EmbeddingComposite
+
+# Create Q matrix
+Q = defaultdict(int)
+
+Q[(1,1)] = -91
+Q[(1,2)] = 72
+Q[(1,3)] = 72
+Q[(2,2)] = -87
+Q[(2,3)] = 72
+Q[(3,3)] = -89
+
+print("\nQUBO:\n")
+for i in range(1,4):
+    row = ''
+    for j in range(1,4):
+        if (i,j) in Q:
+            row = row + str(Q[(i,j)])+'\t'
+        else:
+            row = row + str(0) + '\t'
+    print(row)
+
+qubo_model = BQM.from_qubo(Q)
+ising_model = qubo_model.to_ising()
+
+input()
+print("\nConverting QUBO to Ising ...")
+
+print("\nIsing:\n")
+
+for i in range(1,4):
+    row = ''
+    for j in range(1,4):
+        if j<i:
+            row = row + str(0) + '\t'
+        elif j==i:
+            row = row + str(ising_model[0][i]) + '\t'
+        else:
+            row = row + str(ising_model[1][(i,j)]) + '\t'
+    print(row)
+
+input()
+print("\nEmbedding logical problem into physical layout ...")
+
+# Construct logical problem graph
+prob_graph = nx.Graph()
+prob_graph.add_edges_from([(1,2),(2,3),(1,3)])
+
+# Construct an embedding
+embedding = {1:[1], 2:[2], 3:[3,4]}
+
+# Map our Ising model onto the embedding
+qubits = list(i for x in embedding.values() for i in x)
+target = nx.cycle_graph(qubits)
+th, tJ = dwave.embedding.embed_ising(ising_model[0], ising_model[1], embedding, target)
+
+print("\nQMI (unscaled):\n")
+
+for i in range(1,5):
```

Comment:
Because we have a manual embedding in this problem, the entire program would have to be rewritten for a different QUBO - nothing will generalize without redoing the whole program.  I think we can file that as a feature request on this repo for future work.